### PR TITLE
fix logic error of set world rotation/position/scale

### PR DESCRIPTION
**Pull request by minggo**
Re: #

### Changelog

* world rotation/position/scale should be update if the dirty value is not none though passing value equals to default value

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by minggo** [comment]
@cocos-robot run test cases

**Comment by sushanhong** [comment]
@cocos-robot run test cases

File: cocos/scene-graph/node.ts [hunk 1]
```diff
@@ -2365,20 +2365,29 @@ export class Node extends CCObject implements ISchedulable, CustomSerializable {
     public setWorldPosition (val: Vec3 | number, y?: number, z?: number): void {
         const worldPosition = this._pos;
 
+        // Force update may happen in the situation:
+        // - node is create, and added to a scene
```
Comment (dumganhar) [inline cocos/scene-graph/node.ts]: create -> created

**Review by dumganhar — APPROVED** [review]

File: cocos/scene-graph/node.ts (hunk 1 above)
Comment (minggo) [reply cocos/scene-graph/node.ts]: good catch, fixed

File: tests/core/node.test.ts [hunk 2]
```diff
@@ -463,4 +463,63 @@ describe(`Node`, () => {
...
+        let quat = new Quat();
+        Quat.fromEuler(quat, -30, 0, 0);
+        expect(son.getRotation()).toEqual(quat);
+    });
+
+    test ('setWorldScale', ()=> {
+        let parent = new Node();
+
+        // rotate 30 degrees around x axis
+        parent.setScale(2, 3, 4);
```
Comment (dumganhar) [inline tests/core/node.test.ts:515]: Wrong comment: rotate 30 degrees around x axis ?

Comment (minggo) [reply tests/core/node.test.ts:515]: Good catch, fixed in https://github.com/cocos/cocos-engine/pull/17610.

**Pull request merged**